### PR TITLE
Move the check for any or same type assignability as the first check

### PR DESCRIPTION
**Pull request by MaryamZi**
## Purpose
This PR moves the check for any or same type assignability as the first check in the checkCast/isAssignable methods.

### CODE REVIEW

**Comment by Kishanthan** [comment]
@MaryamZi , do we have an associated issue for this PR?

**Comment by MaryamZi** [comment]
@Kishanthan, we do not have an issue. 

This is an improvement to the unification of the isAssignable/checkCast methods done in https://github.com/ballerina-platform/ballerina-lang/pull/9273.

**Comment by Kishanthan** [comment]
Could we check why the windows build failed?

File: bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/CPU.java [hunk 1]
```diff
@@ -3084,10 +3088,12 @@ public static boolean checkCast(BValue rhsValue, BType lhsType) {
         return checkCastByType(rhsType, lhsType);
     }
 
+    private static boolean checkCastToSameOrAnyType(BType rhsType, BType lhsType) {
```
Comment (Kishanthan) [inline bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/CPU.java]: Shall we rename this to boolean related function name?

Comment (MaryamZi) [reply bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/CPU.java]: Fixed with 1a378a5 as discussed.

**Pull request merged**